### PR TITLE
[netcore] Clean up netcore build

### PR DESCRIPTION
**Pull request by marek-safar**

### CODE REVIEW

File: mono/Makefile.am [hunk 1]
```diff
@@ -16,6 +16,10 @@ if ENABLE_NETCORE
 btls_dirs = 
 endif
 
+if ENABLE_NETCORE
+SUBDIRS = eglib arch utils sgen metadata mini
+else
+
```
Comment (vargaz) [inline mono/Makefile.am]: Aren't the profilers needed ?

Comment (marek-safar) [reply mono/Makefile.am]: that's unclear, we'll need to hook into existing profilers setup for netcore

File: netcore/Makefile [hunk 2]
```diff
@@ -51,7 +51,7 @@ TEST_ASSETS_URL = $(FEED_BASE_URL)/corefx-tests/$(NETCORETESTS_VERSION)/$(TESTS_
 BLAME = $(shell git blame ../configure.ac | grep AC_INIT | cut -f1 -d' ')
 VERSTUB = .$(shell git rev-list --count $(BLAME)..HEAD)
 
-all: runtime bcl
+all: bcl
```
Comment (EgorBo) [inline netcore/Makefile]: why `runtime` was removed here?

Comment (steveisok) [reply netcore/Makefile]: @marek-safar we wanted to be able to build netcore from scratch by just using netcore/build.sh.

Comment (marek-safar) [reply netcore/Makefile]: It was causing circular build dependencies. I don't think we can build with just that script. You need to run `autogen` first and then `make`. Let's stay with this setup.

**Pull request merged**